docs/tinymist/frontend/emacs.typ

```diff
@@ -0,0 +1,102 @@
+
+== Features
+<features>
+See #link("https://github.com/Myriad-Dreamin/tinymist#features")[Tinymist Features] for a list of features.
+
+
+== Finding Executable
+<finding-executable>
+To enable LSP, you must install `tinymist`. You can find `tinymist` on:
+
+- Night versions available at #link("https://github.com/Myriad-Dreamin/tinymist/actions")[GitHub Actions];.
+- Stable versions available at #link("https://github.com/Myriad-Dreamin/tinymist/releases")[GitHub Releases];. \
+    If you are using the latest version of
+    #link("https://codeberg.org/meow_king/typst-ts-mode")[typst-ts-mode], then
+    you can use command `typst-ts-lsp-download-binary` to download the latest
+    stable binary of `tinymist` at `typst-ts-lsp-download-path`.
+
+
+You can also compile and install *latest* `tinymist` by #link("https://www.rust-lang.org/tools/install")[Cargo];.
+
+```bash
+cargo install --git https://github.com/Myriad-Dreamin/tinymist --locked tinymist
+```
+
+
+== Setup Server
+<setup-server>
+
+```el
+(with-eval-after-load 'eglot
+  (with-eval-after-load 'typst-ts-mode
+    (add-to-list 'eglot-server-programs
+                 `((typst-ts-mode) .
+                   ,(eglot-alternatives `(,typst-ts-lsp-download-path
+                                          "tinymist"
+                                          "typst-lsp"))))))
+```
+
+== Extra Settings
+<extra-settings>
+=== Configuring LSP Server
+<configuring-lsp-server>
+
+You can either use `eglot-workspace-configuration` or specifying initial
+launch argument for `tinymist`.
+
+==== eglot-workspace-configuration
+<eglot-workspace-configuration>
+
+For example, if you want to export PDF on save:
+
+```el
+  (setq-default eglot-workspace-configuration
+                '(:exportPdf "onSave"))
+```
+
+Here is an example for both configuring `pylsp` and `tinymist`
+
+```el
+  (setq-default eglot-workspace-configuration
+                '(:pylsp (:plugins (:ruff (:enabled t)))
+                         :exportPdf "onSave"))
+```
+
+NOTE: currently, `tinymist` doesn't support having its name as the top-level
+configuration entry, so the following configuration won't work:
+
+```el
+  (setq-default eglot-workspace-configuration
+                '(:pylsp (:plugins (:ruff (:enabled t)))
+                  :tinymist (:exportPdf "onSave")))
+```
+
+You can also have configuration per directory. Be sure to look at the
+documentation of `eglot-workspace-configuration`.
+
+See #link("https://github.com/Myriad-Dreamin/tinymist/tree/main/editors/neovim/Configuration.md")[Tinymist Server Configuration]
+for references.
+
+
+==== Initial Launch Argument
+
+For example:
+
+```el
+(with-eval-after-load 'eglot
+  (with-eval-after-load 'typst-ts-mode
+    (add-to-list 'eglot-server-programs
+                 `((typst-ts-mode) .
+                   ,(eglot-alternatives `(,typst-ts-lsp-download-path
+                                          ("tinymist" "--font-path" "<your-font-path>")
```

Comment:
It changes only one line of the setup in "Setup Server" section. we may merge "Setup Server" and "Initial Launch Argument" section because of that.

"Besides, you can specify additional launch arguments in `eglot-alternatives`, blabla..."

Reply:
I think it would be better to separate the docs. Mostly users don't want to change the default configuration (function like `export-pdf` already exists in Emacs package `typst-ts-mode`), so they don't need to read them.

And I think `launch argument` is always the second way to set configuration when we have `eglot-workspace-configuration`, since it unconditionally set the configuration, without the ability to set per directory.